agent/reconciler/host_reconciler.go

```diff
@@ -28,14 +29,26 @@ import (
 	infrastructurev1beta1 "github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/apis/infrastructure/v1beta1"
 )
 
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+
+//counterfeiter:generate . Installer
+type Installer interface {
+	Install(string, string, string, logr.Logger) error
+	Uninstall(string, string, string, logr.Logger) error
+}
+
+type K8sInstaller struct {
```

Comment:
this is not needed. We don't want to implement `Install` and `Uninstall` ourselves.